src/Core/BinCodec.fs

```diff
@@ -0,0 +1,72 @@
+﻿namespace B2R2
+
+/// Little Endian Base 128 Codec
+module LEB128 =
+  [<Literal>]
+  let Max32LEB128Length = 5
+  [<Literal>]
+  let Max64LEB128Length = 10
+
+  let DecodeUInt32 (bytes: byte []) =
+    let rec decodeLoop (offset, value, currentByte, length) =
+      let newValue = value ||| ((uint32 currentByte &&& 0x7fu) <<< (offset * 7))
+      if currentByte &&& 0x80uy = 0uy || offset = length - 1 then
+        Ok newValue
+      else
+        decodeLoop (offset + 1, newValue, bytes.[offset + 1], length)
+    if bytes.Length = 0 then
+      Error "Invalid buffer length"
+    else
+      let len = if bytes.Length > Max32LEB128Length then Max32LEB128Length else bytes.Length
+      decodeLoop (0, 0u, bytes.[0], len)
+
```

Comment:
The rest of the functions should follow the same direction as above since they are basically repeating. Two more comments/questions follows.

(1) I have an additional question though. From the wikipedia, it seems that there are two encoding format for LEB128: unsigned LEB128 and signed LEB128. But your PR seems to have four different functions: decodeuint32, decodeuint64, etc. Why is it so? I am not an expert of LEB128 and maybe I misunderstood something?

(2) Seems that all the four functions largely overlap. Can you refactor them?

Reply:
Never mind about the (1). I now understand what it is.

Reply:
I'll try to refactor them